zebrad/src/commands/start/sync.rs

```diff
@@ -154,29 +181,30 @@ where
         Ok(())
     }
 
+    #[instrument(skip(self))]
     async fn extend_tips(&mut self) -> Result<(), Report> {
         // Extend Tips 1
         //
         // remove all prospective tips and iterate over them individually
         let tips = std::mem::take(&mut self.prospective_tips);
+        tracing::debug!(?tips, "extending tip set");
 
         let mut download_set = HashSet::new();
         for tip in tips {
             // ExtendTips Step 2
             //
             // Create a FindBlocksByHash request consisting of just the
             // prospective tip. Send this request to the network F times
+            let mut tip_futs = FuturesUnordered::new();
             for _ in 0..self.fanout {
-                let res = self
-                    .peer_set
-                    .ready_and()
-                    .await
-                    .map_err(|e| eyre!(e))?
-                    .call(zn::Request::FindBlocks {
+                tip_futs.push(self.peer_set.ready_and().await.map_err(|e| eyre!(e))?.call(
+                    zn::Request::FindBlocks {
                         known_blocks: vec![tip],
                         stop: None,
-                    })
-                    .await;
+                    },
+                ));
```

Comment:
TIOLI: I'd split this arg for `push` out into its own let statement just so rustfmt likes it better.

Reply:
personally I think the newer formatting is better because it puts all the ceremony on one line and all the content in a block.